src/queries.ts

```diff
@@ -0,0 +1,102 @@
+import * as vscode from "vscode";
+import {
+  type Model,
+  quickInfoRequest,
+  createInlayHint,
+  getLeftMostHintOfLine
+} from "./helpers";
+
+/** Strongly-typed RegExp groups (https://github.com/microsoft/TypeScript/issues/32098#issuecomment-1279645368) */
+type RegExpGroups<T extends string> = (
+  & IterableIterator<RegExpMatchArray>
+  & {
+    groups: { [name in T]: string } | { [key: string]: string };
+  }[]
+);
+
+const twoslashQueryRegex = /^\s*\/\/\s*\^\?/gm; // symbol: ^?
+const inlineQueryRegex = /^[^\S\r\n]*(?<start>\S).*\/\/\s*(?<end>=>)/gm; // symbol: =>
```

Comment:
https://regex101.com/r/6Jb8h2/1 - skips whitespace and ensures that at least one word character exists before the symbol